grails-app/controllers/org/pih/warehouse/product/ProductGroupController.groovy

```diff
@@ -239,34 +234,38 @@ class ProductGroupController {
      * @return
      */
     def addProductToProductGroup = {
-        println "addProductToProductGroup() " + params
-        def productGroup = ProductGroup.get(params.id)
-        if (productGroup) {
-            def product = Product.get(params.product.id)
-            if (product) {
-                productGroup.addToProducts(product)
-                productGroup.save()
-            }
+        Boolean isProductFamily = params.boolean("isProductFamily") ?: false
+        ProductGroup productGroup = null
+        try {
+            productGroup = productGroupService.addProductToProductGroup(params.id, params.product?.id, isProductFamily)
+        } catch (IllegalArgumentException e) {
+            productGroup = ProductGroup.read(params.id)
+            flash.error = e.message
         }
-        render(template: 'products', model: [productGroup: productGroup, products: productGroup.products])
+        render(template: 'products', model: [productGroup: productGroup, products: isProductFamily?productGroup?.siblings:productGroup?.products])
     }
 
     /**
      * Delete product group from database
      */
     def deleteProductFromProductGroup = {
-        println "deleteProductFromProductGroup() " + params
+        Boolean isProductFamily = params.boolean("isProductFamily") ?: false
         def productGroup = ProductGroup.get(params.id)
         def product = Product.get(params?.product?.id)
         if (product && productGroup) {
-            product.removeFromProductGroups(productGroup)
-            productGroup.removeFromProducts(product)
+            if (isProductFamily) {
+                product.productFamily = null
+            }
+            else {
+                product.removeFromProductGroups(productGroup)
+                productGroup.removeFromProducts(product)
+            }
             product.save(flush: true)
         } else {
             response.status = 404
         }
 
-        render(template: 'products', model: [productGroup: productGroup, products: productGroup.products])
+        render(template: 'products', model: [productGroup: productGroup, products: isProductFamily?productGroup?.siblings:productGroup.products])
```

Comment:
Please add spaces around `?` and `:`. For a brief moment, I thought that there should be a dot after `?` and could not understand why.